### ISSUE TITLE
CardField on Android Not Triggering onCardChange Event, sometimes getting "Card details not complete" log

### ISSUE THREAD

**Issue by nitin-swami-mobisoft** [stripe/stripe-react-native#1797]
**Describe the bug**
 I'm experiencing an issue where the onCardChange event in the CardField component is not triggered when entering card details on Android. The same code works fine on iOS. I've confirmed that all necessary configurations (e.g., publishableKey, dependencies, permissions) are correctly set up, but the event doesn't seem to fire on Android devices.
 
**To Reproduce**
1. Set up a React Native project with the latest version of @stripe/stripe-react-native (tested with version 0.40.0 and 0.38.6).
2. Implement the CardField component with the onCardChange event handler as shown below:

<KeyboardAvoidingView
  behavior={Platform.OS === "ios" ? "padding" : null}
  style={{ backgroundColor: COLORS.white, flex: 1 }}
>
      <CardField
        postalCodeEnabled={true}
        placeholders={{ number: "4242 4242 4242 4242", }}
        cardStyle={{ backgroundColor: "#FFFFFF", textColor: "#000000", }}
        style={{ width: "100%",  height: 50, marginVertical: 30, }}
        onCardChange={(cardDetails) => { console.log("cardDetails", cardDetails);}}
        onFocus={(focusedField) => { console.log("focusField", focusedField);}}
      />
      <TouchableOpacity
        style={{ marginTop: 50, alignSelf: "center" }}
        onPress={() => this.submitPayment()}
      >
        <Text style={styles.buttonRegister}>Pay</Text>
      </TouchableOpacity>
    </View>
  </View>
  <ActivityIndicator animating={this.state.animating} />
</KeyboardAvoidingView>;

4. Run the app on Android (both emulator and physical device).
5. Enter a valid card number (4242 4242 4242 4242) and check if the onCardChange event is fired.
6. Expected Result: The event should be triggered, and the card details should be logged to the console.
7. Actual Result: The onCardChange event is not fired on Android. The event works correctly on iOS.

**Environment**:
React Native version: 0.75.2
@stripe/stripe-react-native version: 0.38.6 or 0.40.0
Tested Android OS version: Android Version 11, 12
Stripe publishable key used: [My Publishable Key]

**Steps Taken:**
Cleared Gradle cache: ./gradlew clean
Rebuilt the app: react-native run-android
Tested on multiple Android devices/emulators
Updated @stripe/stripe-react-native to the latest version
Verified compatibility with React Native version

**Comment by justynpollard1** [comment]
Having same issue

**Comment by nitin-swami-mobisoft** [comment]
@justynpollard1 I'm able to resolve this issue by making the **newArchEnabled=false** in gradle.properties, I thnik latest stripe sdk not supporting to react native new architecture.

**Comment by justynpollard1** [comment]
> @justynpollard1 I'm able to resolve this issue by making the **newArchEnabled=false** in gradle.properties, I thnik latest stripe sdk not supporting to react native new architecture.

Yeah I figured it was due to CardField being a now legacy element and not fully supporting the new arch. I have just finished upgrading everything to the new arch and found this issue with android, also found issue with it being focused and unmounting causing an app crash on ios and android, but thats been reporting already. Will be migrating to CustomSheet.
